components/RouteGuard.tsx

```diff
@@ -26,11 +26,12 @@ export default function RouteGuard({ children }: { children: ReactNode }) {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
-  function authCheck(url: string) {
+  async function authCheck(url: string) {
     // redirect to login page if accessing a private page and not logged in
     const publicPaths = ["/login"];
     const path = url.split("?")[0];
-    if (!authService.isLogged && !publicPaths.includes(path)) {
+    const isServerAuth = await checkServerAuth();
+    if ((!authService.isLogged || !isServerAuth) && !publicPaths.includes(path)) {
```

Comment:
looks safe, but just worried that we could be introducing a security issue

Reply:
I was thinking it was helping with catching cases where the client thinks it's logged in (authService.isLogged), but the server disagree (isServerAuth). So assisting with RouteGuard with preventing routing to dashboard as the task mentioned. I'm not sure about the question on slack (should I add the JWT middleware to the new route?), will let Phillip answer since he was tagged ---> ```like someone on the client side overwriting that func to return true , but either way the api wont return any info without a proper token so not a big deal```

Reply:
I'd like to avoid calling the checkAuth method on every single route change. Instead what I think would be useful is to have an `initialAuthCheck` function that is being called in `useEffect()` that would then call the `checkServerAuth`. That way we only do it once every time we open the app and that should be enough to invalidate the JWT